tidb-server/main.go

```diff
@@ -89,163 +139,74 @@ var (
 		})
 )
 
-func main() {
-	tidb.RegisterLocalStore("boltdb", boltdb.Driver{})
-	tidb.RegisterStore("tikv", tikv.Driver{})
-	tidb.RegisterStore("mocktikv", tikv.MockDriver{})
-
-	runtime.GOMAXPROCS(runtime.NumCPU())
+var (
+	cfg     *config.Config
+	storage kv.Storage
+	dom     *domain.Domain
+	svr     *server.Server
+	xsvr    *xserver.Server
+)
 
+func main() {
 	flag.Parse()
 	if *version {
 		printer.PrintRawTiDBInfo()
 		os.Exit(0)
 	}
-	if *skipGrantTable && !hasRootPrivilege() {
-		log.Error("TiDB run with skip-grant-table need root privilege.")
-		os.Exit(-1)
-	}
-
-	ddlLeaseDuration := parseLease(*ddlLease)
-	tidb.SetSchemaLease(ddlLeaseDuration)
-	statsLeaseDuration := parseLease(*statsLease)
-	tidb.SetStatsLease(statsLeaseDuration)
-	ddl.RunWorker = *runDDL
-	tidb.SetCommitRetryLimit(*retryLimit)
-
-	cfg := config.GetGlobalConfig()
-	cfg.Addr = fmt.Sprintf("%s:%s", *host, *port)
-	cfg.LogLevel = *logLevel
-	cfg.StatusAddr = fmt.Sprintf(":%s", *statusPort)
-	cfg.Socket = *socket
-	cfg.ReportStatus = *reportStatus
-	cfg.Store = *store
-	cfg.StorePath = *storePath
-	cfg.SlowThreshold = *slowThreshold
-	cfg.QueryLogMaxlen = *queryLogMaxlen
-	cfg.TCPKeepAlive = *tcpKeepAlive
-	cfg.SSLCAPath = *sslCAPath
-	cfg.SSLCertPath = *sslCertPath
-	cfg.SSLKeyPath = *sslKeyPath
-
-	xcfg := &xserver.Config{
-		Addr:     fmt.Sprintf("%s:%s", *xhost, *xport),
-		Socket:   *socket,
-		LogLevel: *logLevel,
-	}
-
-	// set log options
-	logConf := &logutil.LogConfig{
-		Level: *logLevel,
-	}
-	if len(*logFile) > 0 {
-		logConf.File = logutil.FileLogConfig{
-			Filename:  *logFile,
-			LogRotate: true,
-		}
-	}
-	err := logutil.InitLogger(logConf)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	// Make sure the TiDB info is always printed.
-	level := log.GetLevel()
-	log.SetLevel(log.InfoLevel)
-	printer.PrintTiDBInfo()
-	log.SetLevel(level)
 
-	if joinCon != nil && *joinCon > 0 {
-		plan.JoinConcurrency = *joinCon
-	}
-	plan.AllowCartesianProduct = *crossJoin
+	runtime.GOMAXPROCS(runtime.NumCPU())
 
-	store := createStore()
+	registerStores()
+	loadConfig()
+	overrideConfig()
+	validateConfig()
+	setGlobalVars()
+	setupLog()
+	printInfo()
+	createStoreAndDomain()
+	setupBinlogClient()
+	createServer()
+	setupSignalHandler()
+	setupMetrics()
+	runServer()
+
+	dom.Close()
+	os.Exit(0)
+}
 
-	if *enablePS {
-		perfschema.EnablePerfSchema()
-	}
-	privileges.Enable = *enablePrivilege
-	privileges.SkipWithGrant = *skipGrantTable
-	if *binlogSocket != "" {
-		createBinlogClient()
-	}
+func registerStores() {
```

Comment:
I suggest register store in `func init()`, some tests may not call main() but they need mocktikv.

Reply:
init in which package?